packages/medusa/src/services/fulfillment.ts

```diff
@@ -224,11 +234,34 @@ class FulfillmentService extends TransactionBaseService {
 
       const created = await Promise.all(
         fulfillments.map(async ({ shipping_method, items }) => {
+          await pvInventoryServiceTx.validateInventoryAtLocation(
+            items,
+            locationId!
+          )
+
+          await Promise.all(
+            items.map(async (i) => {
+              await pvInventoryServiceTx.adjustReservationsQuantityByLineItem(
+                i.id,
+                i.variant_id!,
+                locationId!,
+                -i.quantity
+              )
```

Comment:
todo: this method throws an error if no reservation exists - however, it shouldn't block the ability to create a fulfillment. 

I.e., you can go from `OrderPlaced` to `OrderFulfilled` without allocating in between these two states.